### DISCUSSION TITLE
Transitions & parent handling

### DISCUSSION THREAD

**Discussion by winksaville** [AletheiaWareLLC/Wink#3] [General]
I don't see how transition between states nor how to let a parent handle a message.

**Comment by winksaville** [comment]
Now I see GotoState as how to transition :) But how to have the parent handle the current message, but let the subsequent message be handled by the "current" state. In my HSM implementations the receiving of a message is handled by the infrastructure and it calls the "current" state and then handles changing the "current" state and invoking the parent if needed.

**Comment by stuartmscott** [comment]
Correct, `GotoState` transitions to the given state.

Each state as a set of receivers with a pattern and action. If a message is received that matches a pattern, the action is triggered. If no matches are found, the empty receiver (if any) is triggered. If there is no empty receiver, the message is handled by the parent state (if any). If there is no parent state, an unhandled message error is generated.

See this in action with the timer sample: https://github.com/AletheiaWareLLC/Wink/pull/4
